docs/libcurl/opts/CURLOPT_ECH.md

```diff
@@ -0,0 +1,72 @@
+---
+c: Copyright (C) Daniel Stenberg, <daniel.se>, et al.
+SPDX-License-Identifier: curl
+Title: CURLOPT_ECH
+Section: 3
+Source: libcurl
+See-also:
+  - CURLOPT_DOH_URL (3)
+Protocol:
+  - HTTPS
```

Comment:
At one level, TLS would be more correct. However, for now, ECH support that fetches ECHConfig values (the public keys needed) from the DNS (via DoH), only supports use of the HTTPS RR, which is intended only for HTTPS. At the command-line or library level, if an application finds and supplies the right ECHConfig value, then that should work ok, but I've only done tests of that with web servers on the other side, those being the only ECH-enabled servers with which I can currently test. (Well, haproxy is a little different, not being a web server, but the only protocol tested so far was still HTTPS.)

I'd be happy to test with some other protocol on top of TLS if/when possible.

In summary, I'd say for now, HTTPS is more accurate but it could sensibly change to TLS in future.

Reply:
Ok, thanks for the additional info. There is however a subtle extra detail that I think still argues for this to me marked as `TLS` in this file: we have extra special magic for options that are for TLS, as they also need to specify for which TLS backends they work - which then by extension ends up visible in a [table of TLS options on the webpage](https://curl.se/libcurl/c/tls-options.html).

In that regard this is a TLS option that depends on the specific TLS backend if it is supported or not in curl.

The caveat that this feature probably only works for HTTPS for the moment should probably then be added in the description.

Of course, this feature has the additional complexity documentation wise that it also requires a patch on OpenSSL to work with it, but maybe we can add that in text and yet claim OpenSSL support in the meta-data...

Reply:
Sure, I'll make it say "TLS" and "OpenSSL, boringssl, WolfSSL" and try craft text to explain the actual status. I guess that's ok as it'd also appear as experimental on the web page you mention? Or can say that in text.

Reply:
made that change, will push shortly